R/mlClassificationNeuralNetwork.R

```diff
@@ -0,0 +1,265 @@
+  .errorHandlingClassificationAnalyses(dataset, options, type = "neuralnet")
+  
+  # Check if analysis is ready to run
+  ready <- .classificationAnalysesReady(options, type = "neuralnet")
+  
+  # Determine activation and loss function for the neural net
+  .getNeuralNetworkActFunction(options, jaspResults)
+  # .getNeuralNetworkLossFunction(options, jaspResults)
+  
+  # Compute results and create the model summary table
+  .classificationTable(dataset, options, jaspResults, ready, position = 1, type = "neuralnet")
+  
+  # If the user wants to add the classes to the data set
+  .classificationAddClassesToData(dataset, options, jaspResults, ready)
+  
+  # Add test set indicator to data
+  .addTestIndicatorToData(options, jaspResults, ready, purpose = "classification")
+  
+  # Create the data split plot
+  .dataSplitPlot(dataset, options, jaspResults, ready, position = 2, purpose = "classification", type = "neuralnet")
+  
+  # Create the confusion table
+  .classificationConfusionTable(dataset, options, jaspResults, ready, position = 3)
+  
+  # Create the class proportions table
+  .classificationClassProportions(dataset, options, jaspResults, ready, position = 4)
+  
+  # Create the validation measures table
+  .classificationEvaluationMetrics(dataset, options, jaspResults, ready, position = 5)
+  
+  # Create the network weights table
+  .neuralNetworkTable(dataset, options, jaspResults, ready, purpose = "classification", position = 6)
+  
+  # Create the error plot
+  .nnErrorPlot(dataset, options, jaspResults, ready, position = 7, purpose = "classification")
+  
+  # Create the ROC curve
+  .rocCurve(dataset, options, jaspResults, ready, position = 8, type = "neuralnet")
+  
+  # Create the Andrews curves
+  .classificationAndrewsCurves(dataset, options, jaspResults, ready, position = 9)
+  
+  # Create the activation function plot
+  .neuralNetworkActivationFunctionPlot(options, jaspResults, position = 10)
+  
+  # Create the network graph
+  .neuralNetworkGraph(dataset, options, jaspResults, ready, purpose = "classification", position = 11)
+  
+  # Decision boundaries
+  .classificationDecisionBoundaries(dataset, options, jaspResults, ready, position = 12, type = "neuralnet")
+  
+}
+
+.neuralnetClassification <- function(dataset, options, jaspResults) {
+  
+  # Import model formula from jaspResults
+  formula <- jaspResults[["formula"]]$object
+  
+  # Split the data into training and test sets
+  if (options[["holdoutData"]] == "testSetIndicator" && options[["testSetIndicatorVariable"]] != "") {
+    # Select observations according to a user-specified indicator (included when indicator = 1)
+    train.index <- which(dataset[,options[["testSetIndicatorVariable"]]] == 0)
+  } else {
+    # Sample a percentage of the total data set
+    train.index <- sample.int(nrow(dataset), size = ceiling( (1 - options[['testDataManual']]) * nrow(dataset)))
+  }
+  trainAndValid <- dataset[train.index, ]
```

Comment:
Wouldn't `traindAndValidData` be more descriptive?

Reply:
Same